### PR TITLE
Adds a Simple PowerOutput task and cleans up some of the code.

### PR DESCRIPTION
**Pull request by BlueWeabo**
Features:
- Task for outputting energy to another block (As that is how the energy network functions)
- Deprecate IGlobalWirelessEnergy interface in favor of WirelessNetworkManage class with static implementations
- Improve the documentation of PowerLogic and a few other classes.

### CODE REVIEW

File: src/main/java/gregtech/api/interfaces/tileentity/IEnergyConnected.java [hunk 1]
```diff
@@ -79,28 +83,95 @@ public static long emitEnergyToNetwork(long aVoltage, long aAmperage, IEnergyCon
...
+            }
+            return usedAmperes;
+        }
+
+        /**
+         * Same as {@link #emitEnergyToNetwork(long, long, IEnergyConnected)}, but instead we remove the energy directly from the logic itself.
+         * @param emitter The host, which is trying to emit energy in the network
+         * @param outputSide side from where energy is being outputted to. If its {@link ForgeDirection#UNKNOWN} then it doesn't emit energy to the network
+         */
+        public static void emitEnergyToNetwork(@Nonnull final PowerLogicHost emitter, @Nonnull final ForgeDirection outputSide) {
```
Comment (miozune) [inline src/main/java/gregtech/api/interfaces/tileentity/IEnergyConnected.java:115]: This method has too many duplicated code with the another one

File: src/main/java/gregtech/api/logic/PowerLogic.java [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
-import gregtech.api.interfaces.IGlobalWirelessEnergy;
 
-public class PowerLogic implements IGlobalWirelessEnergy {
+/**
+ * Power logic for machines. This is used to store all the important variables for a machine to have energy and use it
+ * in any way.
+ * 
+ * @author BlueWeabo, Maxim
+ */
+public class PowerLogic {
```
Comment (miozune) [inline src/main/java/gregtech/api/logic/PowerLogic.java:19]: I'm not satisfied with current `PowerLogic` - `PowerOutputTask` - `PowerLogicHost` structure, we might need decision on how to design energy handling. I'll think more tomorrow

Comment (BlueWeabo) [reply src/main/java/gregtech/api/logic/PowerLogic.java:19]: We probably should decide upon it yeah. There is a reason I am trying to not spread it out too much. What we could do is try to go with making the energy network some form of node network instead, but that also has its complications and breaking everything. I would like to hear what you think tho. Should prolly talk about it in #multiblock-madness or somewhere.

File: src/main/java/gregtech/api/interfaces/tileentity/IEnergyConnected.java (hunk 1 above)
Comment (BlueWeabo) [reply src/main/java/gregtech/api/interfaces/tileentity/IEnergyConnected.java:115]: What if I made the other basically have a wrapper class for IEnergyConnected to work for this method? The wrapper class would make its own "powerLogic" and make it just have a "infinite" stored energy, but the same amps and voltage. And this method will still return the amperage, but it won't be needed if directly using a MuTE.

File: src/main/java/gregtech/common/misc/WirelessNetworkManager.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+        // Mark the data as dirty and in need of saving.
+        try {
+            GlobalEnergyWorldSavedData.INSTANCE.markDirty();
+        } catch (Exception exception) {
+            System.out.println("COULD NOT MARK GLOBAL ENERGY AS DIRTY IN ADD EU");
+            exception.printStackTrace();
+        }
+
+        // Get the team UUID. Users are by public static in a team with a UUID equal to their player UUID.
```
[the remark this answers is not in the document]
Comment (BlueWeabo) [reply src/main/java/gregtech/common/misc/WirelessNetworkManager.java:74]: ctrl+f then replace all. my bad there

File: src/main/java/gregtech/api/interfaces/IGlobalWirelessEnergy.java [hunk 4]
```diff
@@ -1,61 +1,44 @@
...
-import gregtech.common.misc.GlobalEnergyWorldSavedData;
+import gregtech.common.misc.WirelessNetworkManager;
 
 // If you are adding very late-game content feel free to tap into this interface.
 // The eventual goal is to bypass laser/dynamo stuff and have energy deposited directly from ultra-endgame
 // multi-blocks directly into the users network.
+/**
+ * Use WirelessNetworkManager instead
+ */
+@Deprecated
```
[the remark this answers is not in the document]
Comment (BlueWeabo) [reply src/main/java/gregtech/api/interfaces/IGlobalWirelessEnergy.java:17]: Sure I shall.

File: src/main/java/gregtech/api/interfaces/tileentity/IEnergyConnected.java (hunk 1 above)
Comment (miozune) [reply src/main/java/gregtech/api/interfaces/tileentity/IEnergyConnected.java:115]: Will it be used outside of this class? I think it's too complicated compared to what we want to achieve.

File: src/main/java/gregtech/api/logic/PowerLogic.java (hunk 2 above)
Comment (miozune) [reply src/main/java/gregtech/api/logic/PowerLogic.java:19]: Actually we do have node system, but it's not used very well. If we could re-rewrite enet (maybe based on CEu) that would be great, but at least we should merge such changes to master in order to catch bugs earlier.

File: src/main/java/gregtech/api/logic/PowerLogic.java [hunk 5]
```diff
@@ -27,43 +34,64 @@ public class PowerLogic implements IGlobalWirelessEnergy {
...
     public PowerLogic setEnergyCapacity(long energyCapacity) {
         this.energyCapacity = energyCapacity;
         return this;
     }
 
+    /**
+     * Sets the maximum amount of amps a machine can receive from an emitter
+     */
     @Nonnull
     public PowerLogic setAmperage(long amperage) {
```
Comment (miozune) [inline src/main/java/gregtech/api/logic/PowerLogic.java:59]: `setMaxAmperage`

**Review by miozune — APPROVED** [review]

**Pull request merged**